pdns/pdnsutil.cc

```diff
@@ -68,6 +68,519 @@ namespace {
+  "Other/Miscellaneous",
+  {{"b2b-migrate", {true, B2BMigrate,
+    "OLD NEW",
+    "\tMove all data from one backend to another"}},
+   {"backend-cmd", {true, backendCmd,
+    "BACKEND CMD [CMD...]",
+    "\tPerform one or more backend commands"}},
+   {"backend-lookup", {true, backendLookup,
+    "BACKEND NAME [[TYPE] CLIENT_IP_SUBNET]",
+    "\tPerform a backend lookup of NAME, TYPE (defaulting to ANY) and\n"
+    "\tCLIENT_IP_SUBNET"}},
+   {"bench-db", {true, benchDb,
+    "[FILENAME]",
+    "\tBenchmark database backend with queries, one zone per line"}},
+   {"create-bind-db", {true, createBindDb,
+    "FILENAME",
+    "\tCreate DNSSEC db for BIND backend (bind-dnssec-db)"}},
+   {"hash-password", {true, hashPassword,
+    "[WORK FACTOR]",
+    "\tAsk for a plaintext password or API key and output a salted and hashed\n"
+    "\tversion"}},
+#ifndef HAVE_P11KIT1 // [
+   {"hsm", {false, HSM,
+    "", ""}}, // not functional so hide it
+#endif // ]
+   {"ipdecrypt", {false, ipDecrypt,
+    "IP_ADDRESS PASSPHRASE_OR_KEY [key]",
+    "\tDecrypt IP address using passphrase or base64 key"}},
+   {"ipencrypt", {false, ipEncrypt,
+    "IP_ADDRESS PASSPHRASE_OR_KEY [key]",
+    "\tEncrypt IP address using passphrase or base64 key"}},
+   {"list-algorithms", {false, listAlgorithms,
+    "[with-backend]",
+    "\tList all DNSSEC algorithms supported, optionally also listing the\n"
+    "\tcryptographic library used"}},
+   {"lmdb-get-backend-version", {false, lmdbGetBackendVersion,
+    "",
+    "\tGet schema version supported by backend"}},
+   {"raw-lua-from-content", {true, rawLuaFromContent,
+    "TYPE CONTENT",
+    "\tDisplay record contents in a form suitable for dnsdist's\n"
+    "\t`SpoofRawAction`"}},
+   {"test-algorithm", {false, testAlgorithm,
+    "ALGONUM",
+    ""}}, // TODO: short help line
+   {"test-algorithms", {false, testAlgorithms,
+    "",
+    ""}}, // TODO: short help line
+   {"test-schema", {true, testSchema,
+    "ZONE",
+    "\tTest DB schema - will create ZONE"}},
+   {"test-speed", {true, testSpeed,
+    "ZONE NUM_CORES",
+    ""}}, // TODO: short help line
+   {"verify-crypto", {true, verifyCrypto,
+    "FILENAME",
+    ""}}} // TODO: short help line
+};
+
+// clang-format on ]
+
+using commandDispatcher = std::map<std::string_view, std::pair<bool, std::vector<groupCommandDispatcher>>>;
+
+static const commandDispatcher topLevelDispatcher{
+  {"autoprimary", {true, {autoprimaryCommands}}},
+  {"catalog", {true, {catalogCommands}}},
+#ifdef HAVE_P11KIT1 // [
+  {"hsm", {true, {HSMCommands}}},
+#endif // ]
+  {"meta", {false, {metadataCommands}}}, // sugar
+  {"meta-data", {false, {metadataCommands}}}, // sugar
+  {"metadata", {true, {metadataCommands}}},
+  {"network", {true, {networkCommands}}},
+  {"record", {false, {rrsetCommands}}}, // sugar
+  {"rrset", {true, {rrsetCommands}}},
+  {"tsig-key", {false, {TSIGKEYCommands}}}, // sugar
+  {"tsigkey", {true, {TSIGKEYCommands}}},
```

Comment:
I wonder if `tsig` is better here. It certainly is shorter, but perhaps not as clear.